src/Caller.cpp

```diff
@@ -707,17 +645,270 @@ bool Caller::parseOptions(int argc, char **argv) {
       cerr << "\nInvoke with -h option for help.\n";
       return 0; // ...error
     }
+
+    decoy_prefix = detect_decoy_prefix(inputFN_);
+    
   }
   // if there is more then one argument left...
   if (cmd.arguments.size() > 1) {
-    cerr << "Error: too many arguments.";
-    cerr << "\nInvoke with -h option for help\n";
-    return 0; // ...error
+    /* Check decoy-prefix */
+    /* TODO: Unnecessary to check for prefix in all files? */
+    for(const string &file : cmd.arguments) {
+
+      decoy_prefix = detect_decoy_prefix(file);
+
+      if (decoy_prefix=="error") {
+        std::cerr << "Couldn't parse decoy prefix from input-file!";
+        return 0;
+      }
+      break;
+      
+    };
+    
+    for(const string &file : cmd.arguments) {
+      if(!detect_tab(file))
+     {
+         std::cerr<< file << " is not comma delimited!\n";
+         return 0;
+     } 
+    };
+  
+    tabInput_ = true;
+
+
+    std::ifstream pinFileStream;
+
+    
+    /* Complete column position to header name */
+    std::map<int,std::string> columnMap = {};
+    /* Temporary mapper while searching for complete position-to-header mapper */
+    std::map<int,std::string> tmpMap;
+
+    /* Keep track on col index */
+    int column_index;
+
+    /* Loop through all files to search for column headers */
+    for(const string &text : cmd.arguments) {
+      
+      /* Get header */
+      pinFileStream.open(text.c_str(), ios::in);
+      std::string hederRow;
+
+      getline(pinFileStream, hederRow);
+
+      TabReader reader(hederRow);
+
+      column_index = 0;
+      while (!reader.error()) {
+        std::string optionalHeader = reader.readString();
+
+        tmpMap[column_index] = optionalHeader;
+        column_index++;
+      }
+
+      /* Get header from file with the most columns */
+      if (columnMap.size() < tmpMap.size()) {
+        columnMap = tmpMap;
+      }
+      tmpMap.clear();
+
+      pinFileStream.close();
+    }
+
+    /* File to write all pin files to, and then feed to Percolator */
+    ofstream outFile;
+    inputFN_ = "tmp_conc_files.tmp";
```

Comment:
Find a file in a temp directory instead. You do not want to be dependent on a given filename.

Reply:
Fixed in: https://github.com/percolator/percolator/commit/c6db156753f4c0d6671de74832afb10831431797